test-karate/src/test/java/graphql/ftm/helpers.feature

```diff
@@ -0,0 +1,120 @@
+Feature: Reusable Functions and Helpers
+
+  Scenario: Define reusable functions
+
+    ## Builds a payload for creating a new content version
+    * def buildContentRequestPayload =
+      """
+      function(contentType, title, publishDate, expiresOn, identifier) {
+        var payload = {
+          "contentlets": [
+            {
+              "contentType": contentType,
+              "title": title,
+              "host":"8a7d5e23-da1e-420a-b4f0-471e7da8ea2d"
+            }
+          ]
+        };
+        if (publishDate) payload.contentlets[0].publishDate = publishDate;
+        if (expiresOn) payload.contentlets[0].expiresOn = expiresOn;
+        if (identifier) payload.contentlets[0].identifier = identifier;
+        return payload;
+      }
+      """
+    ## Extracts all errors from a response
+    * def extractErrors =
+      """
+      function(response) {
+        var errors = [];
+        var results = response.entity.results;
+        if (results && results.length > 0) {
+          for (var i = 0; i < results.length; i++) {
+            var result = results[i];
+            // Handle both nested error messages and direct error messages
+            for (var key in result) {
+              if (result[key] && result[key].errorMessage) {
+                errors.push(result[key].errorMessage);
+              }
+            }
+          }
+        }
+        return errors;
+      }
+      """
+
+    ## Extracts all contentlets from a response
+    * def extractContentlets =
+      """
+      function(response) {
+        var containers = response.entity.containers;
+        var allContentlets = [];
+        for (var key in containers) {
+          if (containers[key].contentlets) {
+            for (var contentletKey in containers[key].contentlets) {
+              allContentlets = allContentlets.concat(containers[key].contentlets[contentletKey]);
+            }
+          }
+        }
+        return allContentlets;
+      }
+      """
+
+    ## Generates a random suffix for test data
+    * def testSuffix =
+      """
+      function() {
+        if (!karate.get('testSuffix')) {
+          var prefix = '__' + Math.floor(Math.random() * 100000);
+          karate.set('testSuffix', prefix);
+        }
+        return karate.get('testSuffix');
+      }
+      """
+
+    ## Extracts a specific object from a JSON array by UUID
+    * def getContentletByUUID =
```

Comment:
if this function can be used by any test and is not limited to contentlets, I would rename it

Reply:
I think the function name is fine